### PR TITLE
Mp/test area

### PR DESCRIPTION
**Pull request by modernpacifist**

### CODE REVIEW

**Review by tmshv — CHANGES_REQUESTED** [review]
has unnecessary bin

**Comment by modernpacifist** [comment]
> has unnecessary bin

you mean __pycache__?

**Review by tmshv — CHANGES_REQUESTED** [review]
It has already committed binaries

**Comment by modernpacifist** [comment]
Can you specify the files?

**Review by tmshv — COMMENTED** [review]
now looks fine
but do nothing

**Comment by modernpacifist** [comment]
Okay, no problem.

File: math_lib.py [hunk 1]
```diff
@@ -7,10 +7,12 @@ def get_radians(degrees):
     return radians
 
 
-def get_coordinates(radians):
-    """"""
-    return math.sin(radians)
+def get_coordinates(direction, radians):
+    """gets direction for the vector"""
+    print(math.tan(get_radians(radians)))
+    return direction * math.tan(get_radians(radians))
```
Comment (tmshv) [inline math_lib.py]: `math.radians`

File: paint_lib.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    pattern_origin_x = test_pattern[1]
+    pattern_origin_y = test_pattern[2]
+    line_shift_along = test_pattern[3]
+    # line_shift_perpendicular = test_pattern[4]
+    # line_length = test_pattern[5]
+    # gap_length = test_pattern[6]
+
+    default_direction = (img_size[0], 0)
+    rotated_direction = math_lib.get_coordinates(img_size[0], 45)
+
```
Comment (tmshv) [inline paint_lib.py:25]:
```py
angle = math.radians(45)
length = img_width
x = math.cos(angle) * length
y = math.sin(angle) * length
```

File: pattern.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Pattern:
+    def __init__(self, angle = 0, x = 0, y = 0, along = 0, perpendicular = 0, length = 0, gap = 0):
+        self.angle = angle
+        self.x = x
+        self.y = y
+        self.line_shift_along = along
+        self.line_shift_perp = perpendicular
+        self.line_length = length
+        self.gap_length = gap
```
Comment (tmshv) [inline pattern.py]: `self.dash = [...]`

**Pull request closed without merging**